nifi-extension-bundles/nifi-box-bundle/nifi-box-processors/src/main/java/org/apache/nifi/processors/box/GetBoxFileCollaborators.java

```diff
@@ -0,0 +1,222 @@
+
+            if (e.getResponseCode() == 404) {
+                getLogger().warn("Box file with ID {} was not found.", fileId);
+                session.transfer(flowFile, REL_NOT_FOUND);
+            } else {
+                getLogger().error("Failed to retrieve Box file collaborations for file [{}]", fileId, e);
+                session.transfer(flowFile, REL_FAILURE);
+            }
+        } catch (final BoxAPIException e) {
+            flowFile = session.putAttribute(flowFile, ERROR_MESSAGE, e.getMessage());
+            flowFile = session.putAttribute(flowFile, ERROR_CODE, String.valueOf(e.getResponseCode()));
+            flowFile = session.penalize(flowFile);
+            session.transfer(flowFile, REL_FAILURE);
+        }
+    }
+
+    /**
+     * Creates a BoxFile instance for a given file ID.
+     *
+     * @param fileId the ID of the file
+     * @return BoxFile instance
+     */
+    @VisibleForTesting
+    protected BoxFile getBoxFile(final String fileId) {
+        return new BoxFile(boxAPIConnection, fileId);
+    }
+
+    private FlowFile fetchCollaborations(final String fileId,
+                                         final ProcessSession session,
+                                         final FlowFile flowFile) {
+        final BoxFile boxFile = getBoxFile(fileId);
+
+        final Iterable<BoxCollaboration.Info> collaborations = boxFile.getAllFileCollaborations();
+
+        final Map<String, List<String>> collabMap = new HashMap<>();
+        collabMap.put("accepted.users", new ArrayList<>());
+        collabMap.put("accepted.groups", new ArrayList<>());
+        collabMap.put("pending.users", new ArrayList<>());
+        collabMap.put("pending.groups", new ArrayList<>());
+        collabMap.put("rejected.users", new ArrayList<>());
+        collabMap.put("rejected.groups", new ArrayList<>());
+
+        int count = 0;
+        for (final BoxCollaboration.Info collabInfo : collaborations) {
+            count++;
+
+            // Determine if collaborator is a user or group
+            boolean isUser = collabInfo.getAccessibleBy().getType().equals(BoxCollaborator.CollaboratorType.USER);
+            final String collabId = collabInfo.getAccessibleBy().getID();
+            final String status = collabInfo.getStatus().toString().toLowerCase();
+            final String type = isUser ? "users" : "groups";
+            final String key = "%s.%s".formatted(status, type);
+
+            if (collabMap.containsKey(key)) {
+                collabMap.get(key).add(collabId);
+            }
+        }
+
+        final Map<String, String> attributes = new HashMap<>();
+        attributes.put(ID, fileId);
+        attributes.put("box.collaborations.count", String.valueOf(count));
+
+        collabMap.forEach((key, value) -> {
+            if (!value.isEmpty()) {
+                addAttributeIfNotEmpty(attributes, "box.collaborations." + key + ".ids", value);
+            }
+        });
```

Comment:
No need to check value twice.

```suggestion
        collabMap.forEach((key, value) -> {
            addAttributeIfNotEmpty(attributes, "box.collaborations." + key + ".ids", value);
        });
```